File: web_frontend/src/api/humanLocks.ts
import { apiGet, apiPost } from "./client";

export type HumanLockItem = {
  id: string;
  shop_id: string;
  buyer_id: string;
  session_id: string;
  status: string;
  reason: string;
  intent: string;
  source: string;
  trace_id: string;
  last_message_preview: string;
  locked_at: string;
  updated_at: string;
  can_unlock: boolean;
};

export type HumanLockListResponse = {
  items: HumanLockItem[];
  total: number;
  shop_id: string;
  warning?: string | null;
};

export type HumanLockUnlockResult = {
  id: string;
  shop_id: string;
  unlocked: boolean;
  already_unlocked: boolean;
  operator: string;
  reason: string;
  updated_at: string;
};

export type HumanLockBatchUnlockResult = {
  shop_id: string;
  matched_count: number;
  unlocked_count: number;
  operator: string;
  reason: string;
};

export type HumanLockListParams = {
  shop_id: string;
  buyer_id?: string;
  session_id?: string;
  status?: string;
  limit?: number;
  offset?: number;
};

function toQuery(params: Record<string, string | number | undefined>) {
  const search = new URLSearchParams();
  Object.entries(params).forEach(([key, value]) => {
    if (value !== undefined && value !== "") search.set(key, String(value));
  });
  return search.toString();
}

export function listHumanLocks(params: HumanLockListParams) {
  return apiGet<HumanLockListResponse>(`/api/human-locks?${toQuery(params)}`);
}

export function unlockHumanLock(id: string, payload: { shop_id: string; operator: string; reason: string }) {
  return apiPost<HumanLockUnlockResult>(`/api/human-locks/${encodeURIComponent(id)}/unlock`, payload);
}

export function unlockHumanLocksBySession(payload: {
  shop_id: string;
  buyer_id?: string;
  session_id?: string;
  operator: string;
  reason: string;
}) {
  return apiPost<HumanLockBatchUnlockResult>("/api/human-locks/unlock-by-session", payload);
}

export function bulkUnlockHumanLocks(payload: { shop_id: string; operator: string; reason: string; limit: number }) {
  return apiPost<HumanLockBatchUnlockResult>("/api/human-locks/bulk-unlock", payload);
}

File: web_api/schemas/human_locks.py
from pydantic import BaseModel, Field


class HumanLockItem(BaseModel):
    id: str
    shop_id: str
    buyer_id: str
    session_id: str
    status: str
    reason: str = ""
    intent: str = ""
    source: str = "unknown"
    trace_id: str = ""
    last_message_preview: str = ""
    locked_at: str = ""
    updated_at: str = ""
    can_unlock: bool = True


class HumanLockListResponse(BaseModel):
    items: list[HumanLockItem]
    total: int
    shop_id: str
    warning: str | None = None


class HumanLockUnlockRequest(BaseModel):
    shop_id: str | None = None
    operator: str = "local_admin"
    reason: str = "manual_unlock_after_review"


class HumanLockUnlockResponse(BaseModel):
    id: str
    shop_id: str
    unlocked: bool
    already_unlocked: bool
    operator: str
    reason: str
    updated_at: str


class HumanLockUnlockBySessionRequest(BaseModel):
    shop_id: str | None = None
    buyer_id: str | None = None
    session_id: str | None = None
    operator: str = "local_admin"
    reason: str = "manual_unlock_after_review"


class HumanLockBulkUnlockRequest(BaseModel):
    shop_id: str | None = None
    operator: str = "local_admin"
    reason: str = "bulk_unlock_after_deploy"
    limit: int = Field(default=100, ge=1, le=500)


class HumanLockBatchUnlockResponse(BaseModel):
    shop_id: str
    matched_count: int
    unlocked_count: int
    operator: str
    reason: str
